packages/flutterfire_cli/lib/src/commands/reconfigure.dart

```diff
@@ -0,0 +1,434 @@
+/*
+
+* Copyright (c) 2016-present Invertase Limited & Contributors
+
+*
+
+* Licensed under the Apache License, Version 2.0 (the "License");
+
+* you may not use this library except in compliance with the License.
+
+* You may obtain a copy of the License at
+
+*
+
+* http://www.apache.org/licenses/LICENSE-2.0
+
+*
+
+* Unless required by applicable law or agreed to in writing, software
+
+* distributed under the License is distributed on an "AS IS" BASIS,
+
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+
+* See the License for the specific language governing permissions and
+
+* limitations under the License.
+
+*
+
+*/
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:path/path.dart' as path;
+
+import '../common/strings.dart';
+import '../common/utils.dart';
+
+import '../firebase.dart';
+import '../firebase/firebase_android_options.dart';
+import '../firebase/firebase_apple_options.dart';
+import '../firebase/firebase_configuration_file.dart';
+import '../firebase/firebase_dart_options.dart';
+import '../firebase/firebase_options.dart';
+import '../flutter_app.dart';
+
+import 'base.dart';
+
+class ConfigFileWrite {
+  ConfigFileWrite({
+    required this.pathToConfig,
+    required this.projectId,
+    this.android,
+    this.ios,
+    this.macos,
+    this.web,
```

Comment:
Maybe we could rename this 
```suggestion
    this.androidOptions,
    this.iosOptions,
    this.macosOptions,
    this.webOptions,
```